src/bindings/python/flux/job/executor.py

```diff
@@ -514,14 +568,36 @@ def submit(self, *args, **kwargs):
             occurred and new jobs cannot be submitted (e.g. a remote Flux instance
             can no longer be communicated with).
         """
+        return self._create_future(_SubmitPackage, args, kwargs)
+
+    def attach(self, jobid):
+        """Attach a ``FluxExecutorFuture`` to an existing job ID and return it.
+
+        Returned futures will behave identically to futures returned by the
+        ``FluxExecutor.submit`` method.
+
+        This method is primarily useful for monitoring jobs that have been
+        submitted through other mechanisms.
+
+        :param jobid: jobid to attach to.
```

Comment:
Comment from a super quick pass between meetings: can this be (does it have to be) a `flux.job.JobID`?  Either way, can you add the type to the function signature?

Reply:
No restrictions on type, but if the type isn't accepted by `flux.job.event_watch_async` the future will error out. I added `:type jobid: int` since I think that's the most general type accepted.